sw/device/lib/dif/README.md

```diff
@@ -31,16 +31,26 @@ contradict with the guidelines below.
 The guidelines below apply to writing DIFs, and code should be written in a
 similar style to the existing DIF libraries in this directory.
 
+### Definitions
+
+<a name="side-effects"></a>
+**Side-effects** include (but are not limited to) writing to memory, including
+memory-mapped hardware, and modifying processor CSRs. This means returning early
+from a guard statement must not leave the hardware in an invalid or
+unrecoverable state.
```

Comment:
This is really minor but I am wondering if the second sentence should be more specific. What do you think about the following?

"If the state of the hardware, i.e. the processor and all peripherals, is _S_ at the beginning of the function and _S'_ at the point of early return, then _S_ and _S'_ can differ only in those components that change indirectly due to the code that executes until the early return, e.g. timer counters, program counter, etc."

Reply:
Two things:
- I think that the second sentence here is actually not part of the definition of side effects, it's about guard statements, and I got it in the wrong place, I'll push it further down the file again (to the rules of guard statements).
- I don't want to be as formal as your proposed sentence - I don't think it helps to think about this abstractly. I prefer to talk about the kinds of actions that are considered to change peripherals in a way that we don't want, rather than whitelisting a set of actions that are allowed.

Reply:
> * I think that the second sentence here is actually not part of the definition of side effects, it's about guard statements, and I got it in the wrong place, I'll push it further down the file again (to the rules of guard statements).

This clears things up, thanks!

Reply:
The most recent update was to address the issue with the side-effects definition.